models/track_and_reid_model.py

```diff
@@ -119,7 +120,16 @@ def get_reid_score_all_ids_full_gallery(track_im_conf, simmat, g_pids):
     aligned_simmat = simmat
     for pid in set(g_pids):
         # maybe take the top-K?
-        ids_score[pid] +=  aligned_simmat[:,np.where(g_pids==pid)].max(axis=1).mean()
+        id_matrix = aligned_simmat[:, np.where(g_pids==pid)]
+        if len(id_matrix) > 0:
+            if len(id_matrix.shape) == 3:
```

Comment:
Instead of checking the shape of the returned tensor, you can simply always call squeeze without defining the axis to remove. This will remove the extra axis if it exists and if not it won't do anything. This will work only if the other channels are not of size 1 in the returned tensor, but correct me if I'm wrong, the problem here was that the returned tensor is of size (x,1,y) where x and y are > 1.
See https://numpy.org/doc/stable/reference/generated/numpy.squeeze.html